src/codeu/chat/server/BroadCastSystem.java

```diff
@@ -0,0 +1,148 @@
+package codeu.chat.server;
+
+import codeu.chat.common.*;
+import codeu.chat.util.Serializers;
+import codeu.chat.util.connections.Connection;
+import codeu.chat.util.Uuid;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.LinkedBlockingQueue;
+
+/**
+ * Created by rsharif on 3/17/17.
+ */
+
+
+public class BroadCastSystem {
+
+    private class ConversationMessageLink {
+
+        Uuid conversationUuid;
+        Message message;
+
+        ConversationMessageLink(Uuid conversationUuid, Message message) {
+            this.conversationUuid = conversationUuid;
+            this.message = message;
+        }
+
+    }
+
+    private ConcurrentHashMap<Integer, List<Connection>> conversationUsers;
+    private final BlockingQueue<ConversationMessageLink> messagesToBroadcast;
+    private final Thread broadCaster;
+
+    private class MessageBroadCaster implements Runnable {
+        @Override
+        public void run() {
+
+            try {
+                while (true) {
+                    broadCastMessage(messagesToBroadcast.take());
+                }
+            } catch (InterruptedException exc) {
+                System.out.println("Error broadcasting message");
+            }
+
+        }
+    }
+
+    public BroadCastSystem(){
+        conversationUsers = new ConcurrentHashMap<>();
+        messagesToBroadcast = new LinkedBlockingQueue<>();
+
+        MessageBroadCaster messageBroadCaster = new MessageBroadCaster();
+
+        broadCaster = new Thread(messageBroadCaster);
+
+        broadCaster.start();
+    }
+
+    private void addConnection(Connection connection, Uuid uuid){
+
+        if (connection == null) throw new NullPointerException();
+        List<Connection> recipients = conversationUsers.get(uuid.id());
+        recipients.add(connection);
+
+    }
+
+    private void removeConnection(Connection connection, Uuid uuid){
+
+        if (connection == null) throw new NullPointerException();
+        List<Connection> recipients = conversationUsers.get(uuid.id());
+        recipients.remove(connection);
+
+    }
+
+    public void switchConversation(Connection connection, ConversationSummary oldCon, ConversationSummary newCon){
+
+        if (newCon == null && oldCon == null)
+            throw new NullPointerException();
+        if (oldCon != null)
+            removeConnection(connection, oldCon.id);
+        if (newCon != null)
+            addConnection(connection, newCon.id);
```

Comment:
Yes, this method will allow them to join more than one conversation at a time. The client side prevents this though. 

Would you suggest I add another check here? If I was to add such a check, I would use another HashMap that maps Connection to Conversation Uuid.